Polymake-0.4 news (#123)
oscar-system/oscar-website: news/_posts/2020-04-27-Polymake.jl-0.4-has-been-tagged.md

---
layout: post

> title: Polymake.jl-0.4 has been tagged

Reviewer: ```suggestion
title: Polymake.jl 0.4 has been tagged
```
(for consistency with other news)

author: Marek Kaluba
---

`Polymake.jl` v0.4.0 has been tagged!

Since we're upgrading to support `CxxWrap.jl` `v0.10.1`, we decided to bump the major 

> version to force compilation of libpolymake against libcxxwrap-julia. The release has been somewhat hastened by the need for for `Oscar.jl` jointly working with the newest `CxxWrap.jl`, so expect rather `v.0.4.1` and later minor versions rather heavy in added features :)

Reviewer: ```suggestion
version to force compilation of libpolymake against libcxxwrap-julia. The release has been somewhat hastened by the need for `Oscar.jl` jointly working with the newest `CxxWrap.jl`, so expect `v.0.4.1` and later minor versions rather heavy in added features :)
```

> !!! Note:

Reviewer: It looks like Jekyll's markdown doesn't support admonitions! Maybe simply:
```suggestion
_Note_:
```
?

---
layout: post
title: Polymake.jl 0.4 has been tagged
author: Marek Kaluba
---

> `Polymake.jl` v0.4.0 has been tagged!

Reviewer 2: "tagged" is technical jargon. Not everybody reading our website will be familiar with that. It also makes it unclear whether the code has also been released.

I would suggest to use here (and in the `title:` field above) something that's easier to understand for those people; e.g. "has ben released" or "is available now" or "is ready" or so.

Reviewer 2: Also, in the `title` it says `0.4.0` and here `v0.4.0`. Pick one?

Reviewer 2: I'd also suggest to use a hyperlink to Polymake.jl so that people can quickly find the repo. Alternatively, it could be to your documentation, if that is available online?

> Since we're upgrading to support `CxxWrap.jl` `v0.10.1`, we decided to bump the major 

Reviewer 2: ```suggestion
Since we're upgrading to support `CxxWrap.jl` v0.10.1, we decided to bump the major 
```
The version is not code, so no need to put it into backticks. It's fine to do use backticks for package names (of course that's a stylistic matter in the end)

version to force compilation of libpolymake against libcxxwrap-julia. 

> The release has been somewhat hastened by the need for `Oscar.jl` jointly working with 

Reviewer 2: ```suggestion
The release has been somewhat hastened by the need for [`Oscar.jl`](https://github.com/oscar-system/Oscar.jl) jointly working with 
```

> the newest `CxxWrap.jl`, so expect `v.0.4.1` and later minor versions rather heavy

Reviewer 2: ```suggestion
the newest `CxxWrap.jl`, so expect `v0.4.1` and later minor versions rather heavy
```

> in added features :)

Reviewer 2: ```suggestion
in added features :).
```

Reviewer: I actually don't find the period great here, it interferes with the interpretation of `:)` as an emoji. I personally am fine with using an emoji as closing punctuation, but would otherwise find it clearer as "features. :)" (quite subjective I know ;-) )
cf. https://rfourquet.github.io/oscar-website/news/2020/04/27/Polymake.jl-0.4-has-been-tagged/

> <br/>

Reviewer 2: Why the explicit line break?

Reviewer: Without, I found it looks too dense, and introducing a new paragraph looks nicer.

_Note_:

> We're tracking the version of libcxxwrap-julia used, so that we can warn You

Reviewer 2: ```suggestion
We're tracking the version of libcxxwrap-julia used, so that we can warn you
```

Reviewer: I was interpreted the capital as an emphasis, with respect to the "We" at the beginning :)

> that `import Pkg; Pkg.build("Polymake")` is necessary, but some of the cases

Reviewer 2: ```suggestion
whenever executing `import Pkg; Pkg.build("Polymake")` is necessary, but some of the cases
```

might have slipped us. If you experience crashes while `using Polymake` try:
 * `import Pkg; Pkg.free("CxxWrap")`,

>  * `import Pkg; Pkg.build("Polymake")`

Reviewer 2: ```suggestion
 * `import Pkg; Pkg.build("Polymake")`,
```

>  * filing bug report!

Reviewer 2: ```suggestion
 * [filing a bug report!](https://github.com/oscar-system/Polymake.jl/issues)
```
Let's use these fancy "hyperlinks" to make life easier for people ;-)

Below you can find a summary of the major changes and design decision we have 

> taken in the development of Polymake.jl.

Reviewer 2: ```suggestion
taken in the development of `Polymake.jl`.
```
Consistency

Reviewer: I was wondering on this one, we are not consistent across the website. I will then update other NEWS accordingly.

<br/>
Happy coding,<br/>
on behalf of polymake Team,<br/>
Marek

> # Changes between Polymake-0.3 and Polymake-0.4:

Reviewer 2: ```suggestion
# Changes between Polymake.jl 0.3 and 0.4
```
Avoids confusion between polymake and Polymake.jl

## Library updates

> * `Polymake.jl` can be used now with `CxxWrap` `^0.10.1`

Reviewer 2: ```suggestion
* `Polymake.jl` can now be used with `CxxWrap` 0.10.1 or later
```

* `Polymake.jl` uses `LoadFlint.jl` and latest `SingularBuilder`

## New features

* documentation strings of all `polymake` functions and methods was made 
available in julia 
* handling of polynomials should be much smoother
* polymake banner is suppressed when imported in Oscar.jl

## Fixed notable bugs

> * `libpolymake` no longer requires GLIBCXX_3.4.26, i.e. Polymake.jl is 

Reviewer 2: ```suggestion
* `libpolymake` no longer requires GLIBCXX_3.4.26, i.e. `Polymake.jl` is 
```
Consistency

---
layout: post
title: Polymake.jl 0.4 has been released
author: Marek Kaluba
---

`Polymake.jl` 0.4 has been
[released](https://github.com/oscar-system/Polymake.jl/releases/tag/v0.4.0)!

> Since we're upgrading to support `CxxWrap.jl` v0.10.1, we decided to bump the major 

Reviewer: ```suggestion
Since we're upgrading to support `CxxWrap.jl` 0.10.1, we decided to bump the major 
```

version to force compilation of libpolymake against libcxxwrap-julia. 
The release has been somewhat hastened by the need for
[`Oscar.jl`](https://github.com/oscar-system/Oscar.jl) jointly working with 
the newest `CxxWrap.jl`, so expect 0.4.1 and later minor versions rather heavy
in added features :).

<br/>
_Note_:
We're tracking the version of libcxxwrap-julia used, so that we can warn you
whenever executing `import Pkg; Pkg.build("Polymake")` is necessary, but some of the cases
might have slipped us. If you experience crashes while `using Polymake` try:
 * `import Pkg; Pkg.free("CxxWrap")`,
 * `import Pkg; Pkg.build("Polymake")`,
 * [filing a bug report!](https://github.com/oscar-system/Polymake.jl/issues)

Below you can find a summary of the major changes and design decision we have 
taken in the development of `Polymake.jl`.

<br/>
Happy coding,<br/>

> on behalf of polymake Team,<br/>

Reviewer 2: ```suggestion
on behalf of the polymake Team,<br/>
```

Marek

# Changes between Polymake.jl 0.3 and 0.4

## Library updates

* `Polymake.jl` can now be used with `CxxWrap` 0.10.1 or later
* `Polymake.jl` uses `LoadFlint.jl` and latest `SingularBuilder`

## New features

* documentation strings of all `polymake` functions and methods was made 
available in julia 
* handling of polynomials should be much smoother
* polymake banner is suppressed when imported in `Oscar.jl`

## Fixed notable bugs

* `libpolymake` no longer requires GLIBCXX_3.4.26, i.e. `Polymake.jl` is 
compatible with libstdc++ shipped by julia, even when using very modern Linux 
systems (big thanks to Max Horn for investigating!)

> * polymake doesn't grab all interrupts, so after `using Polymake` one can 

Reviewer 2: Not sure if it is intentionally that sometimes `polymake` is in a code block and sometimes not (there are valid reasons to do that, of course; e.g. for GAP, I'll write "GAP" for the system, but "`gap`" if I refer to the executable). I don't see a pattern here, but of course that doesn't mean there is one. So please take this suggestion strictly optional (similar for "polymake banner" in line 46).
```suggestion
* `polymake` doesn't grab all interrupts, so after `using Polymake` one can 
```
